paprika/tleap.py

```diff
@@ -478,12 +502,22 @@ def remove_waters_manually(self):
         """
     
         cycle = 0
-        max_cycles = 10
+        max_cycles = 100
         waters = self.wat_added_history[-1]
         while waters > self.target_waters:
             # Retrieve excess water residues
             water_surplus = (waters - self.target_waters)
             water_residues = self.list_waters()
+
+            # THIS IS HACKY. But if we've gone > 5 cycles, we're probably
+            # in a ping-pong convergence problem. So we'll try to solve it
+            # by increasing the water_surplus value, manually, and increase
+            # it as we go through more cycles.
+            if cycle > 5:
+                additional_water = int( float(cycle)/5.0 )
```

Comment:
If `cycle = 6`, won't `int(float(cycle)/5.0)` just be `int(6.0/5.0)` which should be 1? I don't think this will do anything until `cycle = 10`, right?

Reply:
Yup, that was my intent.  Increment the `additional_water` and then try it a few times to see if it can get out of the ping-pong thing.  The `int()` effectively does a floor round, so we end up incrementing every 5 cycles.  I'm in full hack mode here.

Reply:
Should use modulo probably, but I can never get those right the first try when in loops ...

Reply:
Going to see if this fixes my problems!